### PR TITLE
Milestone 3: Association

### PR DESCRIPTION
**Pull request by Lizdev-05**
In this milestone, I achieved the following...

- Created classroom, book, and rental classes
- Set attr_accessor and attr_reader on the classes created
- Implemented many - many relationships and belong-to relationship

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def rent_book(rental)
+    @rentals.push(rental)
+    rental.book = self
+  end
```
Comment (codepantha) [inline book.rb]: - [ ] In the `rent_book` method, you don't have to manually push `rental` to the `@rentals` array. This is done automatically by the `Rental` class when an instance of the `Rental` class is created.
Instead, what you should do is pass in the `person` and the `date`, then the book itself should be sent as a parameter to create the Rental. See below 👇 
``` ruby
def rent_book(person, date)
  Rental.new(date, person, self)
end
```

File: person.rb [hunk 2]
```diff
@@ -24,4 +25,9 @@ def can_use_service
   def correct_name
     @name
   end
+
+  def rent_book(rental)
+    @rentals.push(rental)
+    rental.person = self
+  end
```
Comment (codepantha) [inline person.rb]: - [ ] The same changes I suggested for the Book class applies here as well ☝️

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRES CHANGES ♻️ 

## Hi @Lizdev-05 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

- [ ] I noticed that the link to the MIT license is broken. Kindly fix it 🙏 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 1 above)
Comment (Lizdev-05) [reply book.rb]: Okay, thank you for your insight. I will work on it accordingly

File: person.rb (hunk 2 above)
Comment (Lizdev-05) [reply person.rb]: Noted.

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @Lizdev-05,

Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

thumbs-up-agree


## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Gitflow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by Lizdev-05** [comment]
Thank you @Omar-Muhamad for reviewing and approving my code.

**Pull request merged**